### DISCUSSION TITLE
Where is the cowrie.cfg and/or userdb.txt for the cowrie SSH/telnet honeypot located

### DISCUSSION THREAD

**Discussion by da667** [telekom-security/tpotce#983] [Q&A]
Hey there,

I know that this is something of a really basic question, but how do I modify the cowrie SSH honeypot and the credentials that are used to grant a shell? I'm familiar with how cowrie typically works, but I have zero experience with docker, and thus have no idea where the config files are stored in the container OR how to access them in order to modify them.

I saw this discussion earlier:

https://github.com/telekom-security/tpotce/discussions/941

and the answer seemed to imply that the cowrie.cfg file is what I'm looking for.

I located one at /opt/tpot/docker/cowrie/dist/cowrie.cfg

but there are no configuration options that define how authentication is handled. What's more is that I can't seem to find the userdb.txt file to use to define what credentials are actually used to grant a shell.

I guess this discussion boils down to a handful of questions:

**How do I change cowrie's configuration?**
**Where is the cowrie.cfg file that actually defines how the docker container is configured?**
**Where is the userdb.txt file that defines what credentials work for ssh/telnet? If one is NOT defined, WHERE do I put it?**

**Comment by t3chn0m4g3** [comment]
These files are located inside the docker image. If you want your own settings to work, you can do so by using a docker volume. Please see the examples in the [Wiki](https://github.com/telekom-security/tpotce/wiki).

**Reply by jacobholtz** [reply]
Where did you move the examples after the wiki was removed?
